drivers/ethernet/eth_stm32_hal.c

```diff
@@ -65,7 +65,7 @@ static const struct device *eth_stm32_phy_dev = DEVICE_PHY_BY_NAME(0);
 
 #endif
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_ethernet)
```

Comment:
For readability, I think the replacement of `#if SOC_H7 || SOC_H5` with `HAS_COMPAT(h7_eth)` should be split to a separate commit.

Reply:
ok, i will put it in a separate commit